src/serac/physics/tests/serac_solid_sensitivity.cpp

```diff
@@ -0,0 +1,169 @@
+  // to check if computed qoi sensitivity is consistent
+  // with finite difference on the displacement
+  double eps = 1.0E-6;
+  for (int ix = 0; ix < bulkModulus.Size(); ++ix) {
+    // Perturb bulk sensitivity
+    bulkModulus[ix] = bulkModulusValue + eps;
+    solid.advanceTimestep(timestep);
+    mfem::ParGridFunction displacementPlus = solid.displacement().gridFunc();
+
+    bulkModulus[ix] = bulkModulusValue - eps;
+    solid.advanceTimestep(timestep);
+    mfem::ParGridFunction displacementMinus = solid.displacement().gridFunc();
+
+    // Reset to the original bulk modulus value
+    bulkModulus[ix] = bulkModulusValue;
+
+    // Finite difference to compute sensitivity of displacement with respect to bulk modulus
+    mfem::ParGridFunction du_dbulk(&solid.displacement().space());
+    for (int ix2 = 0; ix2 < displacementPlus.Size(); ++ix2) {
+      du_dbulk(ix2) = (displacementPlus(ix2) - displacementMinus(ix2)) / (2.0 * eps);
+    }
+
+    // Compute numerical value of sensitivity of qoi with respect to bulk modulus
+    // by taking the inner product between adjoint load and displacement sensitivity
+    double dqoi_dbulk = adjointLoad(du_dbulk);
+
+    // See if these are similar
+    std::cout << "dqoi_dbulk " << dqoi_dbulk << std::endl;
+    std::cout << "bulkModulusSensitivity " << bulkModulusSensitivity.localVec()(ix) << std::endl;
+    EXPECT_NEAR(dqoi_dbulk, bulkModulusSensitivity.localVec()(ix), 1.0e-4);
+  }
+
+  bulkModulus = bulkModulusValue;
+
+  for (int ix = 0; ix < shearModulus.Size(); ++ix) {
+    // Perturb bulk sensitivity
+    shearModulus[ix] = shearModulusValue + eps;
+    solid.advanceTimestep(timestep);
+    auto displacementPlus = solid.displacement().gridFunc();
+
+    shearModulus[ix] = shearModulusValue - eps;
+    solid.advanceTimestep(timestep);
+    auto displacementMinus = solid.displacement().gridFunc();
+
+    // Reset to the original shear modulus value
+    shearModulus[ix] = shearModulusValue;
+
+    // Finite difference to compute sensitivity of displacement with respect to bulk modulus
+    mfem::ParGridFunction du_dbulk(&solid.displacement().space());
+    for (int ix2 = 0; ix2 < displacementPlus.Size(); ++ix2) {
+      du_dbulk(ix2) = (displacementPlus(ix2) - displacementMinus(ix2)) / (2.0 * eps);
+    }
+
+    // Compute numerical value of sensitivity of qoi with respect to bulk modulus
+    // by taking the inner product between adjoint load and displacement sensitivity
+    double dqoi_dshear = adjointLoad(du_dbulk);
+
+    // See if these are similar
+    std::cout << "dqoi_dshear " << dqoi_dshear << std::endl;
+    std::cout << "shearModulusSensitivity " << shearModulusSensitivity.localVec()(ix) << std::endl;
+    EXPECT_NEAR(dqoi_dshear, shearModulusSensitivity.localVec()(ix), 1.0e-4);
+  }
+}
+
+int main(int argc, char** argv)
+{
+  MPI_Init(&argc, &argv);
+  ::testing::InitGoogleTest(&argc, argv);
+  const auto result = RUN_ALL_TESTS();
```

Comment:
You need to initialize a SLIC logger otherwise all output from through SLIC (axom and serac) will be noops.